galsim/chromatic.py

```diff
@@ -435,6 +438,12 @@ def drawImage(self, bandpass, image=None, integrator='quadratic', **kwargs):
         # determine combined self.wave_list and bandpass.wave_list
         wave_list, _, _ = utilities.combine_wave_list(self, bandpass)
 
+        # If there are photon ops, they'll probably need valid wavelengths, so add
+        # WavelengthSampler as the first op in the list.
+        if 'photon_ops' in kwargs:
+            wave_sampler = WavelengthSampler(self.SED, bandpass)
+            kwargs['photon_ops'] = [wave_sampler] + kwargs['photon_ops']
```

Comment:
Does this make it illegal to pass in a `WavelengthSampler` explicitly?  Or does the second one become a no-op?

Reply:
It wouldn't be a no-op.  But it would be redundant.  What do you think?  Should we check for that and not add the WavelengthSampler if there already is one.  Or warn the user?  Or maybe WavelengthSampler itself should issue a warning if wavelengths are already set and you're overwriting them?

Reply:
Warning-if-wavelengths-are-set sounds good.